org.adempiere.base/src/org/compiere/print/DataEngine.java

```diff
@@ -607,8 +607,14 @@ else if (AD_Reference_ID == DisplayType.PAttribute)
 					if (synonym == null)
 						synonym = display;
 
-					//	=> A.Name AS AName, table.ID,
-					sqlSELECT.append(m_synonym).append(".").append(display).append(" AS ")
+					// IDEMPIERE-6443
+					if ("City".equals(display)) {
+						sqlSELECT.append("COALESCE(").append(m_synonym).append(".").append(display).append(", '') || '.'");
```

Comment:
question - what is this existing "|| '.'" for ?

Reply:
It's old logic. Maybe a long time ago, querying an empty value returned null, so that logic was needed. I just tested it, and an empty value doesn't return null, so I removed that logic as well.

Reply:
@hieplq - I think oracle does that, AFAIR oracle treats empty values as null